drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/builders/ModelMetaDataEnhancer.java

```diff
@@ -17,37 +17,119 @@
+import org.drools.workbench.models.guided.dtable.shared.model.ActionInsertFactCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.ActionSetFieldCol52;
 import org.drools.workbench.models.guided.dtable.shared.model.BaseColumn;
 import org.drools.workbench.models.guided.dtable.shared.model.ConditionCol52;
 import org.drools.workbench.models.guided.dtable.shared.model.GuidedDecisionTable52;
 import org.drools.workbench.models.guided.dtable.shared.model.Pattern52;
 import org.drools.workbench.services.verifier.plugin.client.api.HeaderMetaData;
+import org.uberfire.commons.validation.PortablePreconditions;
 
 public class ModelMetaDataEnhancer {
 
     private final GuidedDecisionTable52 model;
 
-    public ModelMetaDataEnhancer( final GuidedDecisionTable52 model ) {
-        this.model = PortablePreconditions.checkNotNull( "model",
-                                                         model );
+    public ModelMetaDataEnhancer(final GuidedDecisionTable52 model) {
+        this.model = PortablePreconditions.checkNotNull("model",
+                                                        model);
     }
 
     public HeaderMetaData getHeaderMetaData() {
 
         int columnIndex = 0;
-        final Map<Integer, Pattern52> map = new HashMap();
+        final Map<Integer, ModelMetaData> map = new HashMap<>();
+
+        for (final BaseColumn baseColumn : model.getExpandedColumns()) {
+            if (baseColumn instanceof ConditionCol52) {
+                map.put(columnIndex,
+                        new ModelMetaData(model.getPattern((ConditionCol52) baseColumn),
+                                          PatternType.LHS));
+
+            } else if (baseColumn instanceof ActionInsertFactCol52) {
+                final ActionInsertFactCol52 aif = (ActionInsertFactCol52) baseColumn;
+                map.put(columnIndex,
+                        new ModelMetaData(new FactPattern52Adaptor(aif.getFactType(),
+                                                                   aif.getBoundName()),
+                                          PatternType.RHS));
 
-        for ( final BaseColumn baseColumn : model.getExpandedColumns() ) {
-            if ( baseColumn instanceof ConditionCol52 ) {
-                map.put( columnIndex,
-                         model.getPattern( (ConditionCol52) baseColumn ) );
+            } else if (baseColumn instanceof ActionSetFieldCol52) {
+                final ActionSetFieldCol52 asf = (ActionSetFieldCol52) baseColumn;
+                map.put(columnIndex,
+                        new ModelMetaData(new FactPattern52Adaptor(getFactType(asf),
+                                                                   asf.getBoundName()),
+                                          PatternType.RHS));
             }
 
             columnIndex++;
         }
 
-        return new HeaderMetaData( map );
+        return new HeaderMetaData(map);
+    }
+
+    private String getFactType(final ActionSetFieldCol52 asf) {
```

Comment:
I think this should be placed in one of the following:
- `org.drools.workbench.services.verifier.plugin.client.builders.ColumnUtilitiesBase`
- `org.drools.workbench.services.verifier.plugin.client.builders.VerifierColumnUtilities`

Reply:
True. We might need it elsewhere.

Reply:
And while actually doing the move, noticed this might still be the best place for now at least. VerifierColumnUtilities depends on ModelMetaDataEnhancer so moving this creates a cyclic dependency and it is probably not worth it to fix that before we actually need the getFactType method elsewhere.

Reply:
Hmm, not happy to hear, but well, I am fine if there will be no movement.

Reply:
Spend some time on this area due to portable-bug I fixed in this PR and this needs some sort of clean up. The way the VerifierColumnUtilities and ColumnUtilities work are a bit messy. Tried few things, but the fix takes time. 

The fix might be to just do indexing in server side. We will see.